spinnaker/spinnaker_system/google_http_lb_upsert_scenario.py

```diff
@@ -0,0 +1,681 @@
+    '''
+    bs_upsert = copy.deepcopy(self.__proto_upsert)
+    hr = copy.deepcopy(bs_upsert['hostRules'][0])
+    hr['hostPatterns'] = ['added.host1.com', 'added.host2.com']
+    hr['pathMatcher']['pathRules'][0]['paths'] = ['/added/path']
+    bs_upsert['hostRules'].append(hr)
+
+    payload = self.agent.make_json_payload_from_kwargs(
+      job=[bs_upsert],
+      description='Upsert L7 Load Balancer: ' + self.__lb_name,
+      application=self.TEST_APP
+    )
+
+    contract_builder = gcp.GcpContractBuilder(self.gcp_observer)
+    self._add_contract_clauses(contract_builder, bs_upsert)
+
+    return st.OperationContract(
+      self.new_post_operation(title='add host rule',
+                              data=payload, path='tasks'),
+      contract=contract_builder.build()
+    )
+
+
+  def update_host_rule(self):
+    '''Updates a host rule to the url map.
+    '''
+    bs_upsert = copy.deepcopy(self.__proto_upsert)
+    hr = copy.deepcopy(bs_upsert['hostRules'][0])
+    hr['hostPatterns'] = ['updated.host1.com']
+    hr['pathMatcher']['pathRules'][0]['paths'] = ['/updated/path']
+    bs_upsert['hostRules'].append(hr)
+
+    payload = self.agent.make_json_payload_from_kwargs(
+      job=[bs_upsert],
+      description='Upsert L7 Load Balancer: ' + self.__lb_name,
+      application=self.TEST_APP
+    )
+
+    contract_builder = gcp.GcpContractBuilder(self.gcp_observer)
+    self._add_contract_clauses(contract_builder, bs_upsert)
+
+    return st.OperationContract(
+      self.new_post_operation(title='update host rule',
+                              data=payload, path='tasks'),
+      contract=contract_builder.build()
+    )
+
+
+  def update_port_range(self):
+    '''Updates the port range on the forwarding rule.
+    '''
+    bs_upsert = copy.deepcopy(self.__proto_upsert)
+    bs_upsert['portRange'] = '8080'
+
+    payload = self.agent.make_json_payload_from_kwargs(
+      job=[bs_upsert],
+      description='Upsert L7 Load Balancer: ' + self.__lb_name,
+      application=self.TEST_APP
+    )
+
+    contract_builder = gcp.GcpContractBuilder(self.gcp_observer)
+    self._add_contract_clauses(contract_builder, bs_upsert)
+
+    return st.OperationContract(
+      self.new_post_operation(title='update host rule',
+                              data=payload, path='tasks'),
+      contract=contract_builder.build()
+    )
+
+
+  def add_cert(self, certname):
+    '''Add cert to targetHttpProxy to make it a targetHttpsProxy.
+    '''
+    bs_upsert = copy.deepcopy(self.__proto_upsert)
+    bs_upsert['certificate'] = certname
+
+    payload = self.agent.make_json_payload_from_kwargs(
+      job=[bs_upsert],
+      description='Upsert L7 Load Balancer: ' + self.__lb_name,
+      application=self.TEST_APP
+    )
+
+    contract_builder = gcp.GcpContractBuilder(self.gcp_observer)
+    self._add_contract_clauses(contract_builder, bs_upsert)
+
+    return st.OperationContract(
+      self.new_post_operation(title='add cert',
```

Comment:
You also use this function for a test that changes the cert so parameterize the title and pass it into add_cert.